### ISSUE TITLE
state_string always returns "EMPTY"

### ISSUE THREAD

**Issue by ManafHijazi** [Ziggeo/react-ziggeo#82]
state_string always returns "EMPTY" when trying to access it from the component, but the data is different on the network.
How can I access the response from the last request?

**Comment by Bane-D** [comment]
Hi Manaf,

It is possible that you are getting the state string from a cache of some sorts.

Can you tell us how you are getting this through the code?

**Comment by ManafHijazi** [comment]
Hi Bane,

`const onProcessed = (e) => {
    setStateString(e?.stream_data?.state_string);
    setVideoToken(e.video);
  };

 <ZiggeoRecorder
            apiKey={process.env.NEXT_PUBLIC_ZIGGEO_API_TOKEN}
            video={videoToken}
            onRef={(e) => setRecorder(e)}
            onAttached={onAttached}
            onProcessed={onProcessed}
            onUploading={onUploading}
            recordings={+props.question.number_of_retake + 1}
            timelimit={parseInt(props.question.time_limit)}
            countdown={3}
            onRerecord={onRerecord}
            onRecording={onRecording}
            width="100%"
            webrtc={1}
            webrtc_on_mobile
            webrtc_streaming_if_necessary
          />`

**Comment by sambua** [comment]
Hello @ManafHijazi , unfortunately `stream_data` is only sets once and not updating in the application state. But if you want to achieve some logic and need some data please could you share which data you need? Because our Application very reach with lot of features which are also available with additional API requests [ FRONT-API doc ](https://ziggeo.com/docs/sdks/javascript/api) . Will be happy to help with any advice.

**Comment by ManafHijazi** [comment]
Hey @sambua I want the state_string data to be constantly updating.

**Comment by sambua** [comment]
Hello @ManafHijazi here is example how you can interactively get state from our API.

```
    useEffect(() => {
        if (recorder) {
            // Should be a new instance
            setUpdateInstance(false);
            const app = recorder.application;

            recorder.on("processed", (e) => {
                const response = app.videos.get(recorder.get("video"));
                if (typeof response.success !== "undefined") {
                       // Should get state_string == READY & state == 5
                        response.success((data) => {
                            console.log("RESP success in processed >> ", data, data.state_string, data.state);
                        });
                        response.error((err) => {
                            console.log("RESP error in processed >> ", err);
                        });
                }
            }, recorder);

            recorder.on("uploaded", function () {
                const response = app.videos.get(recorder.get("video"));
                if (typeof response.success !== "undefined") {
                    response.success((data) => {
                        // Should get state_string == EMPTY & state == 3
                        console.log("RESP in Upload success >> ", data, data.state_string, data.state);
                    });
                    response.error((err) => {
                        console.log("RESP in Upload error >> ", err);
                    });
                }
            }, recorder);
      }
    }, [recorder]);

**Comment by ntshcalleia** [comment]
Hi @ManafHijazi, I wanted to follow up on your request. We’ve updated our JS SDK code - which is used by `react-ziggeo` under the hood - to automatically update the video and stream data once we get fresh information from the servers. This update is available starting on version 2.39.1 of the `ziggeo-client-sdk`, so you’d need to update your code to use that version instead. This might help: https://stackoverflow.com/questions/15806152/how-do-i-override-nested-npm-dependency-versions
